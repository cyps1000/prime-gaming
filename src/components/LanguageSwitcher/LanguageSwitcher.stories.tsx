/**
 * Storybook
 * @see https://storybook.js.org/docs/react/get-started/introduction
 */
import { Story, Meta } from "@storybook/react";

/**
 * Component Imports
 */
import LanguageSwitcher, { LanguageSwitcherProps } from "./LanguageSwitcher";

export default {
  title: "Components/LanguageSwitcher",
  component: LanguageSwitcher,
} as Meta;

/**
 * Defines the Template
 * @param args LanguageSwitcherProps
 * @returns
 */
const Template: Story<LanguageSwitcherProps> = (args) => (
  <LanguageSwitcher {...args} />
);

/**
 * Default case
 */
export const Default = Template.bind({});

/**
 * Put your component props in here
 *
 * Example:
 * Default.args = {
 *  text: "Hello World"
 * }
 *
 * Assuming that the component expects a prop text that is a string.
 */
Default.args = {};
