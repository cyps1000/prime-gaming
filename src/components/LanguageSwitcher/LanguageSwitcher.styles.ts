/**
 *  Material UI Imports
 * @see https://material-ui.com/styles/basics/
 */
import { makeStyles } from "@material-ui/core/styles";

/**
 * Styles the component
 */
const useStyles = makeStyles((theme) => ({
  flag: {
    width: 23,
    height: 19,
  },
  container: {
    display: "flex",
    alignItems: "center",
    paddingLeft: 10,
  },
  switchFix: {
    paddingLeft: 10,
  },
  switchPrimary: {},
  switchTrack: {
    backgroundColor: theme.palette.common.white,
  },
}));

export { useStyles };
