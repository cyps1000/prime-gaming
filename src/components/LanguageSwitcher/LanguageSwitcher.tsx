/**
 * Imports Material UI Components
 */
import Switch from "@material-ui/core/Switch";

/**
 * Imports Hooks
 */
import { useLanguage } from "../../hooks";

/**
 * Imports the component styles
 */
import { useStyles } from "./LanguageSwitcher.styles";

/**
 * Defines the props interface
 */
export interface LanguageSwitcherProps {
  text?: string;
}

/**
 * Displays the component
 */
const LanguageSwitcher: React.FC<LanguageSwitcherProps> = (props) => {
  const { text } = props;

  /**
   * Gets the active language and language changer
   */
  const { activeLanguage, changeLanguage } = useLanguage();

  /**
   * Gets the component styles
   */
  const classes = useStyles();

  /**
   * Handles changing the language
   */
  const handleLanguageChange = () => {
    if (activeLanguage === "en_en") {
      changeLanguage("en_ro");
    } else {
      changeLanguage("en_en");
    }
  };

  return (
    <div className={classes.themeFix}>
      <img
        className={classes.flag}
        src="https://www.countryflags.com/wp-content/uploads/romania-flag-png-xl.png"
        alt="ro"
      />
      <div>
        <Switch
          checked={activeLanguage === "en_en"}
          onChange={handleLanguageChange}
          color="secondary"
          name="checkedB"
          inputProps={{ "aria-label": "primary checkbox" }}
          classes={{
            colorPrimary: classes.switchPrimary,
            track: classes.switchTrack,
          }}
        />
      </div>
      <img
        className={classes.flag}
        src="https://www.countryflags.com/wp-content/uploads/united-kingdom-flag-png-large.png"
        alt="uk"
      />
    </div>
  );
};

export default LanguageSwitcher;
