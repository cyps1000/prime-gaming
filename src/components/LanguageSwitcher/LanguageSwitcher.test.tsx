/**
 * @see https://testing-library.com/docs/react-testing-library/intro
 * @see https://www.robinwieruch.de/react-testing-library
 */
import { render } from "@testing-library/react";

/**
 * Imports component
 */
import LanguageSwitcher from "./LanguageSwitcher";

/**
 * Mocking the useTranslation hook
 * @see https://github.com/i18next/react-i18next/issues/876
 *
 */
jest.mock("react-i18next", () => ({
  useTranslation: () => ({ t: (key: string) => key }),
}));

/**
 * Default test
 */
describe("LanguageSwitcher", () => {
  it("renders the component", () => {
    // render(<LanguageSwitcher />);
  });
});
