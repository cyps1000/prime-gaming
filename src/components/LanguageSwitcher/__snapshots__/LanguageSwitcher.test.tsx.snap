// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageSwitcher Snapshot Test passes the snapshot test 1`] = `
"<div data-testid=\\"language-switcher\\" class=\\"makeStyles-container-38\\"><img class=\\"makeStyles-flag-37\\" src=\\"https://www.countryflags.com/wp-content/uploads/romania-flag-png-xl.png\\" alt=\\"ro\\">
  <div><span class=\\"MuiSwitch-root\\"><span class=\\"MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-42 MuiSwitch-switchBase MuiSwitch-colorSecondary makeStyles-switchPrimary-40 PrivateSwitchBase-checked-43 Mui-checked\\" aria-disabled=\\"false\\" data-testid=\\"language-switch-input\\"><span class=\\"MuiIconButton-label\\"><input class=\\"PrivateSwitchBase-input-45 MuiSwitch-input\\" type=\\"checkbox\\" value=\\"\\" checked=\\"\\"><span class=\\"MuiSwitch-thumb\\"></span></span><span class=\\"MuiTouchRipple-root\\"></span></span><span class=\\"MuiSwitch-track makeStyles-switchTrack-41\\"></span></span></div><img class=\\"makeStyles-flag-37\\" src=\\"https://www.countryflags.com/wp-content/uploads/united-kingdom-flag-png-large.png\\" alt=\\"uk\\">
</div>"
`;
